### ISSUE TITLE
asset-loader service in tests is problematic

### ISSUE THREAD

**Issue by trentmwillis** [ember-engines/ember-engines#294]
Because we inject the `assetLoader` into the `Router` class, it means you have to `needs` it in every unit test. This is a bad experience for consumers that we should improve.

Original issue: https://github.com/ember-engines/ember-asset-loader/issues/42

**Comment by leizhao4** [comment]
Also, as mentioned by @dustinfarris, `needs` is not an option in integration tests. So now it's not just about experience.

I was told to add `ember-asset-loader` to my engine's `dependencies`. This did fix the tests, however the engine itself is broken. I guess either this error:
```
Uncaught Error: Failed to load asset manifest. For browser environments, verify the meta tag with name “my-engine/config/asset-manifest" is present. For non-browser environments, verify that you included the node-asset-manifest module.
    at Module.callback (http://localhost:4200/assets/vendor.js:148322:11)
    at Module.exports (http://localhost:4200/assets/vendor.js:132:32)
    at Module.reify (http://localhost:4200/assets/vendor.js:153:59)
    at Module.exports (http://localhost:4200/assets/vendor.js:130:10)
    at requireModule (http://localhost:4200/assets/vendor.js:32:18)
    at http://localhost:4200/assets/vendor.js:178096:20
    at Array.forEach (native)
    at exports.default (http://localhost:4200/assets/vendor.js:178093:8)
    at Module.callback (http://localhost:4200/assets/vendor.js:148355:41)
    at Module.exports (http://localhost:4200/assets/vendor.js:132:32)
```
or this error:
```
Error: Failed to create an instance of 'engine:my-engine'. Most likely an improperly defined class or an invalid module export.
    at instantiate (ember.debug.js:1530)
    at lookup (ember.debug.js:1388)
    at Container.lookup (ember.debug.js:1307)
    at Class.lookup (ember.debug.js:31773)
    at Class.buildChildEngineInstance [as _super] (ember.debug.js:4336)
    at Class.buildChildEngineInstance (engine-instance-ext.js:105)
    at Class.superWrapper [as buildChildEngineInstance] (ember.debug.js:39636)
    at Class._constructEngineInstance (router-ext.js:252)
    at router-ext.js:221
    at tryCatch (ember.debug.js:58165)
```

**Comment by trentmwillis** [comment]
I've opened #295 to fix this for unit tests specifically.

@leizhao4 can you provide a reproduction of this failure for an integration test? The entire container should be present for integration tests, so I would not expect this failure to be present there unless something else is going wrong.

**Comment by rwjblue** [comment]
Fixed by #295

**Comment by leizhao4** [comment]
I haven't been able to work on this but the issue is now gone with ember-engine 0.5.0-beta.7.
